Products/Zuul/catalog/model_catalog.py

```diff
@@ -433,6 +436,23 @@ def _get_fields_to_return(self, fields):
         brain_fields = set(fields) if fields else set()
         return list(brain_fields | MANDATORY_FIELDS)
 
+    def cursor_search(self, search_params, context):
+        try:
+            search_params.fields = self._get_fields_to_return(search_params.fields)
+            catalog_results = self.model_index.cursor_search(search_params)
+        except SearchException as e:
+            log.error("EXCEPTION: {0}".format(e.message))
+            self.raise_model_catalog_error("Exception performing search")
+        else:
+            for result in catalog_results:
+                brains = iter(self._parse_catalog_results(result, context))
+                count = result.total_count
```

Comment:
Question: is `total_count` the count of items in the specific result or the total count of all the items in the catalog?

E.g. If there are 1000 items in the catalog and I request a cursor search with a limit of 10 items.  Does `total_count` have the value 1000 or 10?

Reply:
OK, good.  I think we can now take the next step to make the `search` and `cursor_search` methods achieve parity.  

Currently, the `search` method returns an object that is implements the iterator protocol *and* provides the metadata fields like `total_count`.  The `cursor_search` method returns a generator.  I think we can return the same kind of object from `cursor_search` as is returned from `search`.

You won't be able to use the `SearchResults` class as is.  A `CursorSearchResults` class is needed to handle the nested iteration required to produce the catalog results.  Here's an example of the functionality I want to see from `ModelCatalogTool`'s `cursor_search` method:
```
tool = IModelCatalogTool(dmd)
query = "Products.ZenModel.Device.Device"

results = tool.cursor_search(query, limit=100)
log.info("Found %s devices in the catalog", results.count)
for brain in results:
    log.info("Found device %s", brain.getPath())
```
I can already do this using the `search` method.  Let's make `cursor_search` support the same functionality.